src/components/app/components/Navbar.js

```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Avatar,  Dropdown, Layout, Menu } from 'antd';
+
+import AppLink from './AppLink';
+import home from '../../home';
+
+import signin from '../../signin';
+import signup from '../../signup';
+import { LOGO, MARKET_TEXT, NAVBAR, PATHS, USER_AVATAR } from '../constants';
+
+const { Item } = Menu;
+const { Header } = Layout;
+const { AVATAR, SHAPE, SIZE, SOURCE } = LOGO;
+const { MAIN_NAV, NAV_MENU, NAV_MODE, NAV_THEME } = NAVBAR;
+const { CONTACT, HOME, MARKET, PROFILE, SIGN_IN, SIGN_UP } = PATHS;
+const { CONTAINER, ICON_TYPE, SIGN_OUT, USER_DROP_DOWN, USER_PROFILE } = USER_AVATAR;
+
+const { ContactUs } = home.components;
+const { Signin } = signin.components;
+const { Signup } = signup.components;
+
+const UserMenu = (
+    <Menu>
+        <Item key={USER_PROFILE}>
+            <AppLink to={PROFILE} key={PROFILE}>{USER_PROFILE}</AppLink>
+        </Item>
+        <Item key={SIGN_OUT}>{SIGN_OUT}</Item>
+    </Menu>
+);
+
+const items = [
+    <AppLink key={AVATAR} to={HOME}>
+        <Avatar className={AVATAR} src={SOURCE} size={SIZE} shape={SHAPE} />
+    </AppLink>,
+    <AppLink key={MARKET} to={MARKET}>{MARKET_TEXT}</AppLink>,
+    <Signin key={SIGN_IN} />,
+    <Signup key={SIGN_UP} />,
+    <ContactUs key={CONTACT} />,
+];
+
+/*
+ * this is the the navigation bar at the top of the home page
+ */
+export default class Navbar extends React.Component {
+    render() {
+
+        return (
+            <Header className={MAIN_NAV}>
+                <Menu
+                    className={NAV_MENU}
+                    mode={NAV_MODE}
+                    theme={NAV_THEME}
+                >
+                    {
+                        items.map(item => {
+                            const { key } = item;
+                            const className = key === AVATAR ? CONTAINER : '';
```

Comment:
I don't understand the logic behind this statement

Reply:
`line 32-40` we are defining our items (array of components). Each one has a key value assigned to it. This logic as intended to assign a specific class value to the avatar menu item given that it is different from the rest of the items in the menu. `key === AVATAR ? CONTAINER : '';` is a [ternary](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)

Reply:
OK. I actually knew it is a ternary. I am a little confused what line 58 ought to do exactly being that AVATAR and CONTAINER are strings

Reply:
`AVATAR` and `CONTAINER` are variables that are holding strings. They are not strings by default. I am setting `className` to the value that `CONTAINER` is holding is the value if `key` is strictly equal to `AVATAR` else set `className` to an empty string

Reply:
in simple term, you are saying.... hey key are you in all conditions equal to `AVATAR`?
if your answer is yes, return `CONTAINER`, if your answer is no, return `""` (which is empty string).
From this logic, i learnt that we have filtered the class value of the avatar menu item from the rest of the items in the menu.

Reply:
OK!

Reply:
@EngrEric thats exactly correct